compilation failure with text 1.2.5.0 (ghc 9.0.2)

```
Building executable 'hs-tags' for hs-tags-0.1.5.2..
[3 of 3] Compiling Main             ( Main.hs, /Users/adam.bergmark/repos/3rd/unpacks/hs-tags-0.1.5.2/dist-newstyle/build/x86_64-osx/ghc-9.0.2/hs-tags-0.1.5.2/x/hs-tags/noopt/build/hs-tags/hs-tags-tmp/Main.o )

Main.hs:240:40: error:
    • The constructor ‘IncludeSpecs’ should have 3 arguments, but has been given 2
    • In the pattern: IncludeSpecs qs gs
      In a case alternative:
          IncludeSpecs qs gs -> IncludeSpecs qs (optIncludePath opts ++ gs)
      In the ‘includePaths’ field of a record
    |
240 |                                        IncludeSpecs qs gs -> IncludeSpecs qs (optIncludePath opts ++ gs)
    |                                        ^^^^^^^^^^^^^^^^^^
```

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -8,9 +8,9 @@
 #
 # For more information, see https://github.com/haskell-CI/haskell-ci
 #
-# version: 0.13.20210901
+# version: 0.14
 #
-# REGENDATA ("0.13.20210901",["github","hs-tags.cabal"])
+# REGENDATA ("0.14",["github","hs-tags.cabal"])
 #
 name: Haskell-CI
 on:
@@ -20,12 +20,19 @@ jobs:
   linux:
     name: Haskell-CI - Linux - ${{ matrix.compiler }}
     runs-on: ubuntu-18.04
+    timeout-minutes:
+      60
     container:
       image: buildpack-deps:bionic
     continue-on-error: ${{ matrix.allow-failure }}
     strategy:
       matrix:
         include:
+          - compiler: ghc-9.0.2
+            compilerKind: ghc
+            compilerVersion: 9.0.2
+            setup-method: ghcup
+            allow-failure: false
           - compiler: ghc-9.0.1
             compilerKind: ghc
             compilerVersion: 9.0.1
@@ -69,14 +76,18 @@ jobs:
           apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
           if [ "${{ matrix.setup-method }}" = ghcup ]; then
             mkdir -p "$HOME/.ghcup/bin"
-            curl -sL https://downloads.haskell.org/ghcup/0.1.16.2/x86_64-linux-ghcup-0.1.16.2 > "$HOME/.ghcup/bin/ghcup"
+            curl -sL https://downloads.haskell.org/ghcup/0.1.17.3/x86_64-linux-ghcup-0.1.17.3 > "$HOME/.ghcup/bin/ghcup"
             chmod a+x "$HOME/.ghcup/bin/ghcup"
             "$HOME/.ghcup/bin/ghcup" install ghc "$HCVER"
-            "$HOME/.ghcup/bin/ghcup" install cabal 3.4.0.0
+            "$HOME/.ghcup/bin/ghcup" install cabal 3.6.2.0
           else
             apt-add-repository -y 'ppa:hvr/ghc'
             apt-get update
-            apt-get install -y "$HCNAME" cabal-install-3.4
+            apt-get install -y "$HCNAME"
+            mkdir -p "$HOME/.ghcup/bin"
+            curl -sL https://downloads.haskell.org/ghcup/0.1.17.3/x86_64-linux-ghcup-0.1.17.3 > "$HOME/.ghcup/bin/ghcup"
+            chmod a+x "$HOME/.ghcup/bin/ghcup"
+            "$HOME/.ghcup/bin/ghcup" install cabal 3.6.2.0
           fi
         env:
           HCKIND: ${{ matrix.compilerKind }}
@@ -94,13 +105,13 @@ jobs:
             echo "HC=$HC" >> "$GITHUB_ENV"
             echo "HCPKG=$HOME/.ghcup/bin/$HCKIND-pkg-$HCVER" >> "$GITHUB_ENV"
             echo "HADDOCK=$HOME/.ghcup/bin/haddock-$HCVER" >> "$GITHUB_ENV"
-            echo "CABAL=$HOME/.ghcup/bin/cabal-3.4.0.0 -vnormal+nowrap" >> "$GITHUB_ENV"
+            echo "CABAL=$HOME/.ghcup/bin/cabal-3.6.2.0 -vnormal+nowrap" >> "$GITHUB_ENV"
           else
             HC=$HCDIR/bin/$HCKIND
             echo "HC=$HC" >> "$GITHUB_ENV"
             echo "HCPKG=$HCDIR/bin/$HCKIND-pkg" >> "$GITHUB_ENV"
             echo "HADDOCK=$HCDIR/bin/haddock" >> "$GITHUB_ENV"
-            echo "CABAL=/opt/cabal/3.4/bin/cabal -vnormal+nowrap" >> "$GITHUB_ENV"
+            echo "CABAL=$HOME/.ghcup/bin/cabal-3.6.2.0 -vnormal+nowrap" >> "$GITHUB_ENV"
           fi
 
           HCNUMVER=$(${HC} --numeric-version|perl -ne '/^(\d+)\.(\d+)\.(\d+)(\.(\d+))?$/; print(10000 * $1 + 100 * $2 + ($3 == 0 ? $5 != 1 : $3))')
@@ -135,6 +146,10 @@ jobs:
             prefix: $CABAL_DIR
           repository hackage.haskell.org
             url: http://hackage.haskell.org/
+          EOF
+          cat >> $CABAL_CONFIG <<EOF
+          program-default-options
+            ghc-options: $GHCJOBS +RTS -M3G -RTS
           EOF
           cat $CABAL_CONFIG
       - name: versions

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,11 +2,14 @@
 
 Version history.
 
-## 0.1.6 unreleased
+## 0.1.5.3 Andreas Abel 2022-01-10
 
-## 0.1.5.2 (2021-09-10)
+- Fixed #6: build failure with GHC 9.0.2.
+- Tested with GHC 8.0.2 - 9.0.2.
 
-- Support GHC 8.10.7.
+## 0.1.5.2 Andrés Sicard Ramirez 2021-09-10
+
+- `stack.yaml` file for GHC 8.10.7.
 
 ## 0.1.5.1 Andreas Abel 2021-08-22
 

--- a/Main.hs
+++ b/Main.hs
@@ -51,7 +51,7 @@ import GHC.Data.StringBuffer ( hGetStringBuffer )
 import GHC.Driver.Monad      ( GhcT(..), Ghc(..) )
 import GHC.Driver.Phases     ( pattern Cpp, pattern HsSrcFile )
 import GHC.Driver.Pipeline   ( preprocess )
-import GHC.Driver.Session    ( pattern IncludeSpecs, opt_P, parseDynamicFilePragma, sOpt_P, toolSettings )
+import GHC.Driver.Session    ( includePathsGlobal, opt_P, parseDynamicFilePragma, sOpt_P, toolSettings )
 import GHC.Parser            ( parseModule )
 import GHC.Parser.Lexer      ( P(..), ParseResult(..), PState, mkPState, getErrorMessages )
 import GHC.Settings          ( toolSettings_opt_P )
@@ -76,7 +76,7 @@ import SrcLoc         ( mkRealSrcLoc, noLoc, unLoc )
 import StringBuffer   ( hGetStringBuffer )
 
 #if MIN_VERSION_ghc(8,6,1)
-import DynFlags       ( pattern IncludeSpecs )
+import DynFlags       ( includePathsGlobal )
 #endif
 
 #if MIN_VERSION_ghc(8,10,1)
@@ -236,8 +236,9 @@ main = do
 #endif
 
 #if MIN_VERSION_ghc(8,6,1)
-                    , includePaths = case includePaths dynFlags of
-                                       IncludeSpecs qs gs -> IncludeSpecs qs (optIncludePath opts ++ gs)
+                    , includePaths =
+                        let includeSpecs = includePaths dynFlags
+                        in  includeSpecs { includePathsGlobal = optIncludePath opts ++ includePathsGlobal includeSpecs }
 #else
                     , includePaths = optIncludePath opts ++ includePaths dynFlags
 #endif

--- a/cabal.haskell-ci
+++ b/cabal.haskell-ci
@@ -1,3 +1,0 @@
--- on Linux we can use ghcup to setup (some) of jobs
--- hvr-ppa latest are 9.0.1 and 8.10.4
-ghcup-jobs: > 9.0.1 || > 8.10.4 && < 9.0

--- a/hs-tags.cabal
+++ b/hs-tags.cabal
@@ -1,10 +1,10 @@
 name:            hs-tags
-version:         0.1.6
+version:         0.1.5.3
 cabal-version:   >= 1.10
 build-type:      Simple
 license:         MIT
 license-file:    LICENSE
-copyright:       (c) 2005-2021 The Agda Team.
+copyright:       (c) 2005-2022 The Agda Team.
 author:          Ulf Norell, Nils Anders Danielsson, Andrés Sicard-Ramírez, Andreas Abel, Francesco Mazzoli, Paolo G. Giarrusso
 maintainer:      Andreas Abel <andreas.abel@gu.se>
 bug-reports:     https://github.com/agda/hs-tags/issues
@@ -44,7 +44,7 @@ source-repository head
 source-repository this
   type:     git
   location: https://github.com/agda/hs-tags.git
-  tag:      v0.1.5.2
+  tag:      v0.1.5.3
 
 executable hs-tags
   main-is:          Main.hs

--- a/hs-tags.cabal
+++ b/hs-tags.cabal
@@ -24,6 +24,7 @@ tested-with:
   GHC == 8.8.4
   GHC == 8.10.7
   GHC == 9.0.1
+  GHC == 9.0.2
 
 extra-source-files:
   CHANGELOG.md

